### PR TITLE
SALTO-5932: Context options clean code and add UTs

### PR DESCRIPTION
**Pull request by IdoZakk**
Fixed the previous PR, made a code review and added UTs.
---

I fixed some bugs on the way, and hopefully also the 10K. still did not test the 10K, and also there are still several files that I did not check

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/filters/fields/field_context_option_split.ts [hunk 1]
```diff
@@ -68,14 +71,13 @@ const getOptionsInstances = async ({
...
-          pathNaclCase(naclCase(`${invertNaclCase(context.elemID.name)}_${FIELD_CONTEXT_OPTIONS_FILE_NAME}`)),
-        ]
+        optionInstance.path = adjustPath(context, FIELD_CONTEXT_OPTIONS_FILE_NAME)
         return optionInstance
       }),
     )
-  ).filter(values.isDefined)
+  )
+    .filter(values.isDefined)
+    .sort((a, b) => a.elemID.getFullName().localeCompare(b.elemID.getFullName()))
```
Comment (SeggevHaimovich) [inline packages/jira-adapter/src/filters/fields/field_context_option_split.ts:80]: @IdoZakk 
Why did you add this line?

**Review by IdoZakk — COMMENTED** [review]
Some comments

**Review by shayc331 — COMMENTED** [review]
I started to review and left some comments but it is too long and not trivial. please split it to smaller PRs.
